packages/test/test-end-to-end-tests/src/test/directoryEndToEndTests.spec.ts

```diff
@@ -1347,33 +1347,50 @@ describeCompat(
 			expectSubdirsOrder(sharedDirectory3, dirsInOrder, path);
 		}
 
-		async function pauseAllContainersInbound() {
-			await container1.deltaManager.inbound.pause();
-			await container2.deltaManager.inbound.pause();
-			await container3.deltaManager.inbound.pause();
+		async function pauseAllContainersOutbound() {
+			await container1.deltaManager.outbound.pause();
+			await container2.deltaManager.outbound.pause();
+			await container3.deltaManager.outbound.pause();
 		}
 
-		function resumeAllContainersInbound() {
-			container1.deltaManager.inbound.resume();
-			container2.deltaManager.inbound.resume();
-			container3.deltaManager.inbound.resume();
+		function resumeAllContainersOutbound() {
+			container1.deltaManager.outbound.resume();
+			container2.deltaManager.outbound.resume();
+			container3.deltaManager.outbound.resume();
+		}
+
+		/**
+		 * Wait for the message sent by the current container to be sequenced.
+		 */
+		async function waitForContainerSave(c: IContainer) {
+			if (!c.isDirty) {
+				return;
+			}
+			await new Promise<void>((resolve) => c.on("saved", () => resolve()));
 		}
 
 		it("Eventual consistency in ordering with subdirectories creation/deletion", async () => {
 			// Pause to not allow ops to be processed while we maintained them in order.
-			await pauseAllContainersInbound();
+			await pauseAllContainersOutbound();
```

Comment:
I think to make the test more interesting, it does make sense to pause the inbound queues. This makes sure that e.g. container2 doesn't see container1's create of "dir2" when container2 also concurrently creates a "dir2".

Doing this could also help you keep around some of the intermediate state asserts.

But with the current changes, the test does look deterministic. So good improvement there :)